workloads/imagenet/imagenet_jax/submission.py

```diff
@@ -0,0 +1,177 @@
+"""Training algorithm track submission functions for ImageNet."""
+
+from typing import Iterator, List, Tuple, Any
+
+import jax
+import jax.numpy as jnp
+from jax import lax
+import flax
+from flax import optim
+from flax import jax_utils
+import ml_collections
+
+import spec
+from . import config as config_lib
+
+config = config_lib.get_config()
+
+
+
+# flax.struct.dataclass enables instances of this class to be passed into jax
```

Comment:
I think we can delete TrainState (here and in workload.py), and pass around these three attributes separately wherever this is passed; it looks like all the places we use this wrapper class (which I believe is only in update_params), we already have access to the optimizer state and the global step